README.md

```diff
@@ -1,29 +1,549 @@
+}
+```
+
+It returns a following structure:
+```swift
+public struct JunctionResult {
+    public let chaincodes: [Chaincode]
+    public let password: String?
+}
+```
+
+Example:
+```swift
+let derivationPath = ...
+let junctionFactory = JunctionFactory()
+junctionResult = try junctionFactory.parse(path: derivationPath)
+```
+
+Note: there are two types of junction factory that conform to `JunctionFactoryProtocol` but differ in internal implementation:
+`SubstrateJunctionFactory` is used for parsing substrate-based networks derivation paths
+`BIP32JunctionFactory` is used for parsing BIP32 derivation paths
+
+#### Keypair factory
+The library support keypair generation using the following algorithms:
+* Sr25519 (`SR25519KeypairFactory`)
+* Ed25519 (`Ed25519KeypairFactory`)
+* ECDSA (`EcdsaKeypairFactory`)
+* BIP32 (`BIP32KeypairFactory`)
+
+Each algorithm is implemented as a Factory conforming to one of the following protocols, meaning that in any case you can create a keypair from a seed and in some cases you can derive a child keypair from a parent one:
+```swift
+public protocol KeypairFactoryProtocol {
+    func createKeypairFromSeed(_ seed: Data, chaincodeList: [Chaincode]) throws -> IRCryptoKeypairProtocol
+}
+
+public protocol DerivableKeypairFactoryProtocol: KeypairFactoryProtocol {
+    func deriveChildKeypairFromParent(_ keypair: IRCryptoKeypairProtocol,
+                                      chaincodeList: [Chaincode]) throws -> IRCryptoKeypairProtocol
+}
+```
+
+`EcdsaKeypairFactory` and `Ed25519KeypairFactory` also conform to `DerivableSeedFactoryProtocol` so you can derive child seed from a parent seed
+```swift
+public protocol DerivableSeedFactoryProtocol: KeypairFactoryProtocol {
+    func deriveChildSeedFromParent(_ seed: Data,
+                                   chaincodeList: [Chaincode]) throws -> Data
+}
+```
+
+All components usage example:
+```swift
+let junctionFactory = SubstrateJunctionFactory()
+let seedFactory = SeedFactory()
+let keypairFactory = SR25519KeypairFactory()
+
+let path = "//foo/bar"
+let mnemonic = "bottom drive obey lake curtain smoke basket hold race lonely fit walk"
+
+let junctionResult = try junctionFactory.parse(path: path)
+
+let seedResult = try seedFactory.deriveSeed(
+    from: mnemonic,
+    password: junctionResult.password ?? "")
+
+let keypair = try keypairFactory.createKeypairFromSeed(
+    seedResult.seed.miniSeed,
+    chaincodeList: junctionResult.chaincodes
+)
+
+let publicKey = keypair.publicKey().rawData()
+let privateKey = keypair.privateKey().rawData()
+```
 
+### Keystore
+
+
+### Icon
+One of the features of Polkadot UI is a special icon, generated from public key data. Currently, there is TypeScript implementation and we just re-implemented it for iOS. Generally, it just packs smaller circles inside the hexagon inscribed into an outer circle with a radius of 32 points. Colors to fill small circles are chosen based on binary representation of the public key.
```

Comment:
`there is TypeScript implementation...` where?

Reply:
added link